gusto/transport_equation.py

```diff
@@ -38,30 +126,7 @@ def __init__(self, state, V, *, ibp="once", solver_params=None):
         self.test = TestFunction(V)
         self.trial = TrialFunction(V)
 
-        # find out if we are CG
-        nvertex = V.ufl_domain().ufl_cell().num_vertices()
-        entity_dofs = V.finat_element.entity_dofs()
-        # If there are as many dofs on vertices as there are vertices,
-        # assume a continuous space.
-        try:
-            self.is_cg = sum(map(len, entity_dofs[0].values())) == nvertex
-        except KeyError:
-            self.is_cg = sum(map(len, entity_dofs[(0, 0)].values())) == nvertex
-
-        # DG, embedded DG and hybrid SUPG methods need surface measures,
-        # n and un
-        if self.is_cg:
-            self.dS = None
-            self.ds = None
-        else:
-            if V.extruded:
-                self.dS = (dS_h + dS_v)
-                self.ds = (ds_b + ds_t + ds_v)
-            else:
-                self.dS = dS
-                self.ds = ds
-            self.n = FacetNormal(state.mesh)
-            self.un = 0.5*(dot(self.ubar, self.n) + abs(dot(self.ubar, self.n)))
+        self.dS = surface_measures(V)
```

Comment:
What happened to the exterior surface measures (`ds`, `ds_tb`, etc)? Do we not need them?

Reply:
We only use them for outflow boundary conditions which are only used in @tommbendall's rain parameterisations. I thought it best to remove them from a more general function otherwise it looks like we might do something with them more general. Boundary conditions are something I plan to sort out, or at least improve in some way, after FML...